src/lightning/data/streaming/cache.py

```diff
@@ -102,6 +106,20 @@ def filled(self) -> bool:
         self._is_done = os.path.exists(os.path.join(self._cache_dir, _INDEX_FILENAME))
         return self._is_done
 
+    @property
+    def checkpoint_dir(self) -> str:
```

Comment:
It shouldn't be necessary to duplicate the code in both of these.

Reply:
Sure.